pkg/server/router/presentation.go

```diff
@@ -157,19 +158,116 @@ func (pdr PresentationDefinitionRouter) GetPresentationDefinition(ctx context.Co
 	id := framework.GetParam(ctx, IDParam)
 	if id == nil {
 		errMsg := "cannot delete a presentation without an ID parameter"
 		logrus.Error(errMsg)
 		return framework.NewRequestErrorMsg(errMsg, http.StatusBadRequest)
 	}
 
-	if err := pdr.service.DeletePresentationDefinition(presentation.DeletePresentationDefinitionRequest{ID: *id}); err != nil {
+	if err := pr.service.DeletePresentationDefinition(presentation.DeletePresentationDefinitionRequest{ID: *id}); err != nil {
 		errMsg := fmt.Sprintf("could not delete presentation with id: %s", *id)
 		logrus.WithError(err).Error(errMsg)
 		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusInternalServerError)
 	}
 
 	return framework.Respond(ctx, w, nil, http.StatusOK)
 }
+
+type CreateSubmissionRequest struct {
+	SubmissionJWT keyaccess.JWT `json:"submissionJwt" validate:"required"`
+}
+
+type Operation struct {
+	ID     string          `json:"id"`
+	Done   bool            `json:"done"`
+	Result OperationResult `json:"result,omitempty"`
+}
+
+type OperationResult struct {
+	Error    string                          `json:"error,omitempty"`
+	Response exchange.PresentationSubmission `json:"response,omitempty"`
+}
+
+// CreateSubmission godoc
+// @Summary      Create Submission
+// @Description  Creates a submission in this server ready to be reviewed.
+// @Tags         SubmissionAPI
+// @Accept       json
+// @Produce      json
+// @Param        request  body      CreateSubmissionRequest  true  "request body"
+// @Success      201      {object}  Operation
+// @Failure      400      {string}  string  "Bad request"
+// @Failure      500      {string}  string  "Internal server error"
+// @Router       /v1/presentations/submissions [put]
+func (pr PresentationRouter) CreateSubmission(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	var resp Operation
+	return framework.Respond(ctx, w, resp, http.StatusCreated)
+}
+
+type GetSubmissionResponse struct {
+	Submission exchange.PresentationSubmission `json:"submission"`
+}
+
+// GetSubmission godoc
+// @Summary      Get Submission
+// @Description  Get a submission by its ID
+// @Tags         SubmissionAPI
+// @Accept       json
+// @Produce      json
+// @Param        id   path      string  true  "ID"
+// @Success      200  {object}  GetSubmissionResponse
+// @Failure      400  {string}  string  "Bad request"
+// @Router       /v1/presentations/submission/{id} [get]
+func (pr PresentationRouter) GetSubmission(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	var resp GetSubmissionResponse
+	return framework.Respond(ctx, w, resp, http.StatusOK)
+}
+
+type ListSubmissionRequest struct {
+	Filter string `json:"filter"`
+}
+
+type ListSubmissionResponse struct {
+	Submissions []exchange.PresentationSubmission `json:"submissions"`
+}
+
+// ListSubmissions godoc
+// @Summary      List Submissions
+// @Description  List existing submissions according to a filtering query.
```

Comment:
maybe here with a link to the filter syntax we use

Reply:
Added here. Will likely expand on this once I do the implementation.